tracer/test/Datadog.Trace.Tests/OpenTelemetryOperationNameMapperTests.cs

```diff
@@ -0,0 +1,70 @@
+// <copyright file="OpenTelemetryOperationNameMapperTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.Activity;
+using Datadog.Trace.Tagging;
+using Xunit;
+
+namespace Datadog.Trace.Tests
+{
+    [Collection(nameof(OpenTelemetryOperationNameMapperTests))]
+    public class OpenTelemetryOperationNameMapperTests
+    {
+        // Note: These test cases were copy/pasted from parametric tests (with some find/replace to make it work here)
```

Comment:
For reference: https://github.com/DataDog/system-tests/blob/c6c79880e30fb67795d95dd8e1b61932e5aa2420/tests/parametric/test_otel_span_methods.py#L414

Reply:
:chef-kiss: 

One comment I have is AFAICT that these tests don't test the _order_ of the detection when you have "conflicting" tags.

An "alternative" approach you _may_ want to consider is having a span that has "all" the tags, and then check that "removing" a tag moves you onto the next span type.


e.g. 

- `db.System=Redis`
- `http.request.method=GET` 

With both these tags, you get `http.<server|client>.request`. With the `http` tag removed you get `redis.query`.

This would apply to the parametric tests to, and you could potentially "build" the test data from the data you have here for each iteration. Just a thought!

Reply:
Okay I tried doing this in f26a152be016ab792b29766616d1fb2888a4ed8f

I didn't realize that with inherited `Tags` classes you can't access the `RemoveTag` call when setting a value to `null` so it seems a bit hacky as I have to recreate the tags each time I remove one.